src/components/PvGenerator/PvGeneratorComponent.tsx

```diff
@@ -216,6 +302,17 @@ export class PvGeneratorComponent extends React.Component<WidgetProps> {
                     </Tabs>
                 </div>
                 <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} refreshMode={"throttle"} refreshRate={33}></ReactResizeDetector>
+                <ConfirmationDialogComponent
```

Comment:
Maybe it's simpler to use `<Alert>` here (with Yes/No buttons and Cancel by clicking elsewhere). I'm wondering if it's more clear to set a "overwrite generated PV images" toggle in the settings?

Reply:
Maybe we can keep all the pv images by default and we can have a button in the pv generator that clear all the existing pv images without generating new images and when users click on clear all pv images, we alert the users "do you want to clear all the pv images?"

Reply:
Usually we try different pv cuts and explore the results iteratively. That's why we need the ability to keep or remove the generated pv images. The current "alert" with yes/no is a bit interruptive as every time we will be asked once again. I would suggest either we use a toggle or a button in the main widget for keeping or removing existing pv images. As for a toggle or a button, I don't have a preference. 🤔

Reply:
I have added a toggle button. Please review.